### DISCUSSION TITLE
Database trigger or NextJs logic

### DISCUSSION THREAD

**Discussion by laurent512** [prisma/prisma#9209] [Q&A]
Hi everyone,

I am developing a database simulating wallet & transactions behaviors.
When transaction is issued, wallets balances are updated (one goes up, the other goes down of the transaction amount). 
 
Practically, there are two models : 

- wallet :  ID, balance, (some more) , createdAt
- transaction : ID, createdAt, issuingWallet , receivingWallet, amount

To maintain the wallet balances up to date, I am considering two options : 

1. Create database trigger , (each time new transaction is issued, wallets are updated)
2. Update the wallet balance directly using prisma in my nextjs code in the api part

Could you please advice me to take the right decision and let me know the pro & cons of each.
Also, If I take option 1. , can I perform verification on it ? (e.g. transaction is denied if one wallet goes <0€ after trigger)

Thank you so much !! :)

**Comment by ryands17** [comment]
@laurent512 👋 

Updating the wallet balance directly in your API using a transaction should work fine in this case.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
